### PR TITLE
Add rating/review creation endpoint.

### PR DESCRIPTION
**Pull request by jfhamlin**
@lelandg559, @jjfreund, opening this PR for discussion. I'm starting with some tweaks to the ratings table. AFAICT from the mocks in Zeplin, ratings and reviews are left on "resources"__*__. @derekfidler, is this correct?

__*__: _or OpenReferral "locations" — I'm going to get on that rename soon, before we dig ourselves even deeper on "resources"_

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -2,4 +2,16 @@ class ApplicationController < ActionController::API
...
+
+  private
+
+  def require_authorization!
+    head :unauthorized unless current_user
+  end
+
+  def current_user
+    return @current_user if defined? @current_user
+    user_id = request.headers['Authorization']
```
Comment (jfhamlin) [inline app/controllers/application_controller.rb]: This is just to get us off the ground. Once we have proper login support, we'll be providing proper auth tokens here. For now, we'll allow requests to claim to be any user by providing an existing user's ID.

File: spec/requests/resources/ratings/create_spec.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+          'review' => nil
+        }
+      )
+    end
+
+    context 'with a review' do
+      let(:rating_body) do
+        {
+          rating: 3,
+          review: 'Great staff.'
```
Comment (jfhamlin) [inline spec/requests/resources/ratings/create_spec.rb]: It seems to make most sense to present ratings and reviews as combined entities to clients. A single request creates a rating with an optional review.

File: spec/controllers/ratings_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+require 'rails_helper'
+
+RSpec.describe RatingsController, type: :controller do
```
Comment (jfhamlin) [inline spec/controllers/ratings_controller_spec.rb]: If there are no controller tests to be added, this file should be removed.

File: spec/requests/resources/ratings/create_spec.rb (hunk 2 above)
Comment (jfhamlin) [inline spec/requests/resources/ratings/create_spec.rb]: It'd be good to strip/trim review text before creating a review object. An empty string or a whitespace-only string should _not_ create a review.

**Comment by jfhamlin** [comment]
@jjfreund, @lelandg559, I'm contemplating dropping the `reviews` table and just adding a nullable `review` column to `ratings`. It would make the code a bit simpler, and I don't think we'll lose any expressiveness. What do you think?

**Comment by lelandg559** [comment]
ill rebase/fix this up so we can use it.

**Comment by lelandg559** [comment]
after talking with joe, I will add ratings for services.

also reviewing collapsing ratings/reviews into a single table, as suggested

**Comment by jjfreund** [comment]
I'll merge in when the conflict is cleared.

**Pull request merged**